### PR TITLE
Use real now instead of fake when not testing

### PR DESCRIPTION
**Pull request by tedbow**
Talked with @TravisCarden and @effulgentsia  about this. Create a new `Clock` class that we replace for testing

### CODE REVIEW

File: src/Helper/Clock.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Tuf\Helper;
+
+/**
+ * Provides time related functions.
+ */
+class Clock
+{
+
```
Comment (phenaproxima) [inline src/Helper/Clock.php:10]:
```suggestion
```

File: tests/Client/TestUpdater.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use Tuf\Client\RepoFileFetcherInterface;
+use Tuf\Client\Updater;
+use Tuf\Tests\TestHelpers\TestClock;
+
+/**
+ * An updater class for testing uses \Tuf\Tests\TestHelpers\TestClock class.
+ */
+class TestUpdater extends Updater
+{
+
```
Comment (phenaproxima) [inline tests/Client/TestUpdater.php:15]:
```suggestion
```

File: tests/Client/TestUpdater.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+
```
Comment (phenaproxima) [inline tests/Client/TestUpdater.php:3]:
```suggestion
```

File: tests/TestHelpers/TestClock.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        // Use the same timestamp as used in generate_fixtures.py to create
+        // fixtures
+        $this->time = 1577836800;
+    }
+
+    public function getCurrentTime(): int
+    {
+        $this->time+= 5;
+        return $this->time;
```
Comment (phenaproxima) [inline tests/TestHelpers/TestClock.php:24]: Why does this mutate the value of $this->time? If there's a specific reason, can you add a comment explaining it?

**Review by phenaproxima — COMMENTED** [review]
Just a quick request for clarification, plus some whitespace issues.

File: tests/TestHelpers/TestClock.php (hunk 4 above)
Comment (tedbow) [reply tests/TestHelpers/TestClock.php:25]: added a comment

File: tests/Client/TestUpdater.php (hunk 3 above)
Comment (tedbow) [reply tests/Client/TestUpdater.php]: fixed

File: tests/Client/TestUpdater.php (hunk 2 above)
Comment (tedbow) [reply tests/Client/TestUpdater.php]: fixed

File: src/Helper/Clock.php (hunk 1 above)
Comment (tedbow) [reply src/Helper/Clock.php]: fixed

File: tests/Client/TestUpdater.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Tuf\Tests\Client;
+
+use Tuf\Client\RepoFileFetcherInterface;
+use Tuf\Client\Updater;
+use Tuf\Tests\TestHelpers\TestClock;
+
+/**
+ * An updater class for testing uses \Tuf\Tests\TestHelpers\TestClock class.
```
Comment (phenaproxima) [inline tests/Client/TestUpdater.php:10]:
```suggestion
 * An updater class for testing using the \Tuf\Tests\TestHelpers\TestClock class.
```

File: tests/TestHelpers/TestClock.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+namespace Tuf\Tests\TestHelpers;
+
+use Tuf\Helper\Clock;
+
+/**
+ * A test clock class that uses the time from the test fixtures as starting time.
+ */
+class TestClock extends Clock
+{
+    private $time = null;
```
Comment (phenaproxima) [inline tests/TestHelpers/TestClock.php:12]:
```suggestion
    /**
     * The current time.
     *
     * Defaults to the same timestamp used in generate_fixtures.py to create
     * test fixtures.
     */
    private $time = 1577836800;
```

If we use this default value, we can remove __construct().

**Review by phenaproxima — APPROVED** [review]
Approved, just some minor suggestions.

File: tests/Client/TestUpdater.php [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * An updater class for testing using the \Tuf\Tests\TestHelpers\TestClock class.
+ */
+class TestUpdater extends Updater
+{
+    public function __construct(RepoFileFetcherInterface $repoFileFetcher, array $mirrors, \ArrayAccess $durableStorage)
+    {
+        parent::__construct($repoFileFetcher, $mirrors, $durableStorage);
+        $this->clock = new TestClock();
```
Comment (TravisCarden) [inline tests/Client/TestUpdater.php:17]: @tedbow You need to be able to inject a mock of the clock in order to control its behavior. You probably want to add it to the constructor arguments.

Comment (tedbow) [reply tests/Client/TestUpdater.php]: Yep. thanks fixed.

Comment (phenaproxima) [reply tests/Client/TestUpdater.php]: I'm not seeing where this is fixed...?

File: tests/TestHelpers/TestClock.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+{
+    /**
+     * The current time.
+     *
+     * Defaults to the same timestamp used in generate_fixtures.py to create
+     * test fixtures.
+     */
+    private $time = 1577836800;
+
+    public function __construct()
```
Comment (phenaproxima) [inline tests/TestHelpers/TestClock.php]: Now that $this->time has a default value, I don't think we need `__construct()` at all.

File: tests/Client/TestUpdater.php (hunk 7 above)
Comment (tedbow) [reply tests/Client/TestUpdater.php]: @phenaproxima `\Tuf\Tests\Client\TestUpdater::__construct()` now has a `$testClock` argument

Comment (tedbow) [reply tests/Client/TestUpdater.php]: So in the future we could provide `$testClock` that have the time where root is not expired but another file like snapshot.json is.

File: tests/TestHelpers/TestClock.php (hunk 8 above)
Comment (tedbow) [reply tests/TestHelpers/TestClock.php]: Removed it. In the future we will provide want to provide different `$time` values to test `timeshoot.json` being expired but `root.json` not being expired but we can change that when write those tests.

File: tests/TestHelpers/TestClock.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+     *
+     * Defaults to the same timestamp used in generate_fixtures.py to create
+     * test fixtures.
+     */
+    private $time = 1577836800;
+
+    public function getCurrentTime(): int
+    {
+        // Increment the time simulate time passing between calls.
+        $this->time+= 1;
```
Comment (phenaproxima) [inline tests/TestHelpers/TestClock.php:23]:
```suggestion
        $this->time++;
```

File: tests/TestHelpers/TestClock.php [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /**
+     * The current time.
+     *
+     * Defaults to the same timestamp used in generate_fixtures.py to create
+     * test fixtures.
+     */
+    private $time = 1577836800;
+
+    public function getCurrentTime(): int
```
Comment (phenaproxima) [inline tests/TestHelpers/TestClock.php:20]: For correctness, maybe an `@inheritDoc` here?

**Review by phenaproxima — APPROVED** [review]
Approved but two minor things requested. :)

File: tests/TestHelpers/TestClock.php (hunk 10 above)
Comment (tedbow) [reply tests/TestHelpers/TestClock.php:23]: fixed

**Pull request merged**